src/runtime/features.rs

```diff
@@ -0,0 +1,602 @@
+///
+#[derive(
+    Builder,
+    Clone,
+    Debug,
+    Default,
+    Deserialize,
+    Eq,
+    MutGetters,
+    Getters,
+    Setters,
+    PartialEq,
+    Serialize,
+)]
+#[serde(rename_all = "camelCase")]
+#[builder(
+    default,
+    pattern = "owned",
+    setter(into, strip_option),
+    build_fn(error = "OciSpecError")
+)]
+#[getset(get_mut = "pub", get = "pub", set = "pub")]
+pub struct Features {
+    /// The minimum OCI Runtime Spec version recognized by the runtime, e.g., "1.0.0".
+    oci_version_min: String,
+    /// The maximum OCI Runtime Spec version recognized by the runtime, e.g., "1.0.2-dev".
+    oci_version_max: String,
+    /// The list of the recognized hook names, e.g., "createRuntime".
+    /// "None" means "unknown", not "no support for any hook".
+    hooks: Option<Vec<String>>,
+    /// The list of the recognized mount options, e.g., "ro".
+    /// "None" means "unknown", not "no support for any mount option".
+    /// This list does not contain filesystem-specific options passed to mount(2) syscall as (const void *).
+    mount_options: Option<Vec<String>>,
+    /// Information specific to Linux
+    linux: Option<Linux>,
+    /// Implementation-specific annotation strings,
+    /// such as the implementation version, and third-party extensions.
+    annotations: Option<HashMap<String, String>>,
+    /// The list of the potential unsafe annotations
+    /// that may appear in `config.json`.
+    /// A value that ends with "." is interpreted as a prefix of annotations.
+    potentially_unsafe_config_annotations: Option<Vec<String>>,
+}
+
+/// Linux specific features.
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Linux {
+    /// The list of the recognized namespaces, e.g., "mount".
+    /// "None" means "unknown", not "no support for any namespace".
+    namespaces: Option<Vec<String>>,
+    /// The list of the recognized capabilities , e.g., "CAP_SYS_ADMIN".
+    /// "None" means "unknown", not "no support for any capability".
+    capabilities: Option<Vec<String>>,
+
+    cgroup: Option<Cgroup>,
+    seccomp: Option<Seccomp>,
+    apparmor: Option<Apparmor>,
+    selinux: Option<Selinux>,
+    intel_rdt: Option<IntelRdt>,
+    mount_extensions: Option<MountExtensions>,
+}
+
+/// Cgroup represents the "cgroup" field.
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Cgroup {
+    v1: Option<bool>,
+    v2: Option<bool>,
+    systemd: Option<bool>,
+    systemd_user: Option<bool>,
+    rdma: Option<bool>,
+}
+
+/// Seccomp represents the "seccomp" field.
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Seccomp {
+    enabled: Option<bool>,
+    actions: Option<Vec<String>>,
+    operators: Option<Vec<String>>,
+    archs: Option<Vec<String>>,
```

Comment:
Can we use it?
https://github.com/musaprg/oci-spec-rs/blob/6ecc08b6f348c28cd769f3264356f08f95994d77/src/runtime/linux.rs#L1093

Reply:
I've added the missing `ScmpArchRiscv64` to the `Arch` enum.

2325f34

Reply:
fixed to reuse existing types in e247914